theconfluenceBOT/new.py

```diff
@@ -410,27 +420,27 @@ def get_random_posts(reddit):
     Returns a random subreddit according to the algorithm
     """
 
+    sub = None
     rand = random.randint(0, 9)
-
+    
     # 80% chance of random subreddit
     if rand < 8:
         sub = reddit.random_subreddit(nsfw=False)
-        # 50-50 chance of sorting by new or hot
-        hot_or_new = random.randint(0, 1)
-        if hot_or_new == 0:
-            return sub.display_name, sub.hot(limit=10)
-        else:
-            return sub.display_name, sub.new(limit=10)
-
     # 10% chance of r/popular
     elif rand < 9:
         sub = reddit.subreddit("popular")
-        return "popular", sub.new(limit=10)
-
     # 10% chance of r/all
     elif rand < 10:
         sub = reddit.subreddit("all")
-        return "all", sub.hot(limit=10)
+
+    if sub is not None:
+        hot_or_new = random.randint(0, 1)
+        # 50-50 chance of sorting by new or hot
```

Comment:
hot_or_new selection should be specific to random subreddit conditional branch
https://i.imgur.com/lWHrfAN.jpeg

Reply:
is changed back to conditional behaivior